server/src/main/java/org/candlepin/async/tasks/ImportJob.java

```diff
@@ -0,0 +1,226 @@
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (uploadedFileName == null || uploadedFileName.isEmpty()) {
+                    String errmsg = "uploaded file name has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+            }
+            catch (ArgumentConversionException e) {
+                String errmsg = "One or more required arguments are of the wrong type";
+                throw new JobConfigValidationException(errmsg, e);
+            }
+        }
+    }
+
+    private OwnerCurator ownerCurator;
+    private ManifestManager manifestManager;
+
+    @Inject
+    public ImportJob(OwnerCurator ownerCurator, ManifestManager manifestManager) {
+        this.ownerCurator = ownerCurator;
+        this.manifestManager = manifestManager;
+    }
+
+    @Override
+    public Object execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments args = context.getJobArguments();
+
+        String ownerKey = args.getAsString(OWNER_KEY);
+        ConflictOverrides overrides = new ConflictOverrides(args.getAs(CONFLICT_OVERRIDES, String[].class));
+        String storedFileId = args.getAsString(STORED_FILE_ID);
+        String uploadedFileName = args.getAsString(UPLOADED_FILE_NAME);
+
+        Throwable caught;
+        Owner targetOwner = null;
+        try {
+            targetOwner = ownerCurator.getByKey(ownerKey);
+            if (targetOwner == null) {
+                throw new NotFoundException(String.format("Owner %s was not found.", ownerKey));
+            }
+
+            ImportRecord importRecord = manifestManager.importStoredManifest(targetOwner,
+                storedFileId, overrides, uploadedFileName);
+            log.info("Async import complete.");
+            return importRecord;
+        }
+        // Catch and handle SyncDataFormatException and ImporterExceptions
+        // as the OwnerResource.importManifest does which will provide a little more
+        // info about the exception that was thrown (CandlepinException).
+        catch (SyncDataFormatException e) {
+            caught = new BadRequestException(e.getMessage(), e);
+        }
+        catch (ImporterException e) {
+            caught = new IseException(e.getMessage(), e);
+        }
+        catch (Exception e) {
+            caught = e;
+        }
+
+        manifestManager.recordImportFailure(targetOwner, caught, uploadedFileName);
+
+        // If an exception was thrown, the importer's transaction was rolled
+        // back. We want to make sure that the file gets deleted so that it
+        // doesn't take up disk space. It may be possible that the file was
+        // already deleted, but we attempt it anyway.
+        manifestManager.deleteStoredManifest(storedFileId);
+        throw new JobExecutionException(caught.getMessage(), caught, false);
```

Comment:
This, along with the note above, probably requires a bit more thought. I imagine some exceptions will be non-terminal (DB-related ones or otherwise unexpected exceptions), whereas the SyncDataFormat and ImporterExceptions are probably terminal exceptions and should be flagged as such.

Reply:
Fixed.